### ISSUE TITLE
Zoom performance regression in Firefox 49

### ISSUE THREAD

**Issue by geoloep** [Leaflet/Leaflet#4973]
While converting one of my projects from using Leaflet 0.7.7 to Leaflet 1.0, I immediately noticed the performance in Firefox while zooming dropped considerably. Especially zooming in looks very jerky and sometimes ghost images of other zoom levels seem to appear. Zooming in version 0.7.7 is much smoother in Firefox. In Chrome both versions perform very well. (Not surprisingly the problem increases when the map container increases in size). The performance problems are also noticeable on the main site of the Openstreetmap.

## How to reproduce

- Leaflet version I'm using:
0.7.7 & 1.0.0
- Browser (with version) I'm using:
Firefox 49.0
- OS/Platform (with version) I'm using:
Ubuntu 16.04

## What behaviour I'm expecting and which behaviour I'm seeing

In expected version 1.0.0 to at least match the performance of version 0.7.7. In reality performance is poorer.

## Minimal example reproducing the issue

I got the same results using these playgrounds:
[0.7.7](http://playground-leaflet.rhcloud.com/xal)
[1.0.0](http://playground-leaflet.rhcloud.com/faba)

[Example of performance](https://gfycat.com/ParchedDarkHare) (compression may make it harder to see).

I hope this message is of use. I suppose Firefox is not the fastest browser out there, so perhaps what I'm seeing is expected behaviour. In that case I'll keep using 0.7.7. Please let me know if there are other types of information I could provide.

**Comment by IvanSanchez** [comment]
I can't reproduce in my firefox 51.0a2, but I'm gonna make a wild guess and say it's related to the `will-change` CSS rule on the zoom transforms. Maybe the comments in #4959 hold true for this use case.

**Comment by geoloep** [comment]
Small update: On Windows 7 and Firefox 49 the performance of 0.7.7 and 1.0.0 seemed to be mostly similar. On Ubuntu 14.04 and Firefox 49 I the got same results as I had on Ubuntu 16.04 with Leaflet 1.0.0 performing noticeably worse.

**Comment by mourner** [comment]
@geoloep can you remove the `will-change` line in the Leaflet CSS and see how it performs after that?

**Comment by geoloep** [comment]
I wasn't able to detect significant performance differences with the two instances of will-change disabled. I also noticed that for me in Firefox markers do no remain on their intended location during zooming, only after the zoom change is complete they return to their original location. In chrome none of this happens.

I tried to take some performance measurements with the developer tools but it's hard to recreate the same circumstances and compare with measurements from Chrome. I'm pretty confused about this problem, I do not want to send you on a wild goose chase. If it can't be reproduced by some else I feel there remains a chance there is something about my system setup that I'm missing. I of course tried to disable add ons and such.

**Comment by geoloep** [comment]
After some more digging I seem to have tracked down the issue. It appears that most of the time the hardware acceleration of Firefox is disabled by default on Linux. After forcing Firefox to use hardware acceleration anyway the performance of Leaflet 1.0 returned to an expected level. This solution would also explain why I only noticed the performance differential on my Linux systems. I had never considered the status of hardware acceleration in my browser, up till now I didn't experience performance problems like the ones I had with leaflet 1.0, 0.7 apparently didn't require it enabled in my case.

**Comment by IvanSanchez** [comment]
@geoloep How did you check HW accel? It would be worth to investigate if that can be detected in the JS side of things and disable some leaflet features if so.

**Comment by geoloep** [comment]
In Firefox the address "about:support" leads to diagnostic information and showed me the status of HW acceleration (which stated "blocked by default: Acceleration blocked by platform" in my case). Detecting the acceleration in JS and adjusting performance would be cool, I myself am not aware of way's to accomplish that but I'm not experienced in that side of things.

**Comment by IvanSanchez** [comment]
Yeah, it seems that FFX disables HW accel on linux by default, except on nightly builds, see https://reviewboard.mozilla.org/r/59526/

**Comment by IvanSanchez** [comment]
...and it seems that it cannot be detected in a reliable way :sob:  https://www.html5rocks.com/en/tutorials/3d/css/#toc-featuredetection

**Comment by IvanSanchez** [comment]
I think this is the best resource for checking if HW accel is enabled: https://stackoverflow.com/questions/26356769/test-if-hardware-acceleration-has-been-enabled-for-a-css-animation - again, only doable manually by the user, not queryable through JS.

I've given a shot to detecting this by timing the "jankiness" of CSS animations, to no avail.

**Comment by geoloep** [comment]
When you tried to detect the "jankiness" did you do that by calculating the time between requestAnimationframe callbacks? By comparing timestamps it should be possible to calculate the fps of an animation.  I might give that a go but there would be no point if you already determined that's a dead end. Also I'm not so familiar with the leaflet code so I'm not sure I'll even get very far with this.
